runtime/gc_modron_startup/arrayCopy.cpp

```diff
@@ -75,7 +76,11 @@ typeCheckArrayStore(J9VMThread *vmThread, J9Object *object, J9IndexableObject *a
 				return (0 != instanceOfOrCheckCast(storedClazz, componentType));
 			}
 		}
+	} else if (J9_IS_J9CLASS_PRIMITIVE_VALUETYPE(componentType)) {
```

Comment:
The new code should be inside `#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)`

Reply:
added in [92cb616](https://github.com/eclipse-openj9/openj9/commit/92cb6169f6ef83030ce3d192917cf7b963164ba2)